### ISSUE TITLE
[Tag-Editor] Feature

### ISSUE THREAD

**Issue by krfong916** [krfong916/bottomline#6]
❗ 11/9 Spec Change
## What 
- We need a custom tag selector (from herein, referred to as tag-editor or editor) for our question form. Upon consideration of accessibility and extensibility, I've decided that the editor's UX should be based on a combobox with a grid popup [[1]](#1). Bringing inspiration from stackoverflow's [tag editor](https://stackoverflow.com/questions/ask), we want to provide similar functionality.

## Components
- [x] Tag Component
- [x] Combobox Logic: Custom React Hook
- [x] UseCase Component

## Behavior 
- [x] When the tag input is focused, and the user types or pastes text, we fetch tags that are a near match to the input
- [x] When a user types and hits enter, a tag is created
- [x] When a user pastes text, tags created separated by whitespace
- [x] when a tag is created, any `onTagCreated` callbacks are called
- [x] When the user types, the popup of potential suggestions is visible
- [x] A keyclick on a suggested tag closes the popup and creates the tag
- [x] When a tag is created, the input is cleared, the created cannot be updated with keyboard strokes
- [x] A user is able to delete a created tag
- [x] A user is able to update a created tag 

## References
<a id="1">[1]</a> [W3 Combobox Authoring Specification](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/grid-combo.html)

## 🚫  Retired Specification
**From here below is the retired specification**
#### Create Tag
- [x] adding a space in-between a string should create a tag from the beginning of the string up to (but including) the spacebar, and the input should be filled with string from the spacebar index + 1 to the end of the string

#### Update/Edit Tag
- [x] forward edit a tag. 
``` 
Pre-condition: rhs container must not be empty. 

Conditions: if current input value is empty do not create a tag with an empty value. 

If the input is not empty, store the value of the input and the current index

get and store the head of the rhs container 

delete the head of the rhs container

create a new tag with the current input value if it's not empty

use the stored head of the rhs to create the current input value
set state, map, and index
```
#### Delete
- [x] delete: backspace and merge
```
Pre-condition: if the selection is equal to 0, the key is a backspace, the lhs container is not empty, and the current tag is not the head of the lhs container

store the text value from the current input, the value will be concatenated with the string of the previous tag

call editPrev() with the stored text value
```
